libs/payments/paypal/src/lib/paypal.manager.spec.ts

```diff
@@ -38,6 +41,37 @@ describe('paypalManager', () => {
     expect(paypalManager).toBeTruthy();
   });
 
+  describe('getBillingAgreement', () => {
+    it('returns agreement details', async () => {
+      const billingAgreementId = faker.string.sample();
+      const successfulBAUpdateResponse =
+        NVPBAUpdateTransactionResponseFactory();
+
+      paypalClient.baUpdate = jest
+        .fn()
+        .mockResolvedValueOnce(successfulBAUpdateResponse);
+
+      const expected = {
+        city: successfulBAUpdateResponse.CITY,
+        countryCode: successfulBAUpdateResponse.COUNTRYCODE,
+        firstName: successfulBAUpdateResponse.FIRSTNAME,
+        lastName: successfulBAUpdateResponse.LASTNAME,
+        state: successfulBAUpdateResponse.STATE,
+        status: successfulBAUpdateResponse.BILLINGAGREEMENTSTATUS,
```

Comment:
Shouldn't the status value be transformed before returned by `getBillingAgreement`? So shouldn't it be different than the output from `baUpdate`?